velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1399,4 +1400,30 @@ struct TimeZoneMinuteFunction : public TimestampWithTimezoneSupport<T> {
   }
 };
 
+template <typename T>
+struct ToISO8601FromDateFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<Date>& date) {
+    const std::chrono::time_point<std::chrono::system_clock, date::days>
```

Comment:
QQ: How is this logic different from https://github.com/facebookincubator/velox/blob/main/velox/type/Type.cpp#L996 ?

Reply:
Ah, I didn't know of that function. Tried it out - only issue is that it doesn't seem to pad the year to four digits. Realistically probably not a big deal, as it would be unlikely anyone actually inputs data representing years prior to 1000, but maybe this can be fixed separately in the Timestamp parser.

Reply:
@svm1 : Agree that should be fixed in Timestamp parser. It is an existing bug in the logic seems like.

Reply:
@aditi-pandit Realized there is a flag in `TimestampToStringOptions` that dictates zero-padding of the year, enabled it in `DateType::toString()`.